server/index.js

```diff
@@ -0,0 +1,68 @@
+
+const express = require("express");
+
+const app = express();
+app.use(express.json());
+
+const checkBodyValid = (req, res, next) => {
+  if (Object.keys(req.body).length === 0) {
+    return res.status(400).send("Bad request");
+  }
+  next();
+};
+
+const checkToken = (req, res, next) => {
+  if (!req.headers["auth-token"]) {
+    return res.status(401).send("Neautorizat");
+  }
+  next();
+};
+
+app.get("/", (req, res) => {
+  res.send("Yes master, I'm here :)");
+});
+
+app.get("/kittens", checkToken, (req, res) => {
+  res.send(JSON.stringify([
+    {age: 3,
+     name: 'Lilly',
+     breed: 'Siberian',
+     isAdopted: false,
+     photoUrl: 'https://imgix.ranker.com/user_node_img/3386/67703891/original/siberian-freestyle-list-photo-u1?w=650&q=50&fm=pjpg&fit=crop&crop=faces',
+     description:'Strong, alert, and adventurous with lustrous coats and round, curious eyes'
+    },
+    {age: 1,
+     name: 'Miu',
+     breed: 'Somali', 
+     isAdopted: true,
+     photoUrl: 'https://www.askideas.com/media/26/Cute-Somali-Kitten.jpg', 
+     description: 'Highly intelligent, good natured and playful'}]));
+});
+
+app.post("/kittens", checkToken, checkBodyValid, (req, res) => {
+  console.log("new kitty");
+  const kitty = req.body;
+  const randomid=Math.floor(Math.random() * 1000);
+  kitty.id =randomid;
+  res.send(kitty);
+});
+
+app.put("/kittens", checkToken, checkBodyValid, (req, res) => {
+  console.log("update kitty");
+  res.send(req.body);
+});
+app.delete("/kittens/id", checkToken, checkBodyValid,(req,res)=>{
```

Comment:
Pe delete nu se pune body, se trimite doar un id in path
```suggestion
app.delete("/kittens/:id", checkToken, checkBodyValid,(req,res)=>{
```